### PR TITLE
Add fio package

### PR DESCRIPTION
**Pull request by lyoung-confluent**
Fixes:
* #19701

#### For new package PRs only
- [x] This PR is marked as fixing a pre-existing package request bug
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: fio.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - uses: git-checkout
+    with:
+      repository: https://github.com/axboe/fio
+      tag: fio-${{package.version}}
+      expected-commit: 028fcc7db00155fe329497d7501184e8fe49342f
+
+  - runs: |
+      ./configure --prefix=/usr
+      make
+      make DESTDIR="${{targets.destdir}}" install
```
Comment (ajayk) [inline fio.yaml:30]:
```suggestion
      make DESTDIR="${{targets.contextdir}}" install
```

File: fio.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+subpackages:
+  - name: fio-docs
+    pipeline:
+      - uses: split/manpages
+
+update:
+  enabled: true
+  github:
+    identifier: axboe/fio
+    strip-prefix: fio-
```
Comment (ajayk) [inline fio.yaml:41]: Lets add a basic test 
```suggestion
    strip-prefix: fio-
 
test:
    pipeline:
        - name: Verify fio installation
          runs: |
            fio --version || exit 1   
```

File: fio.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/axboe/fio
+      tag: fio-${{package.version}}
+      expected-commit: 028fcc7db00155fe329497d7501184e8fe49342f
+
+  - runs: |
+      ./configure --prefix=/usr
```
Comment (tuananh-vpbank) [inline fio.yaml:28]: could use builtin melange pipelines for consistency?

```yaml
- uses: autoconf/configure
- uses: autoconf/make
- uses: autoconf/make-install
```

Comment (lyoung-confluent) [reply fio.yaml:28]: I tried that initially, but it's actually not an autoconf configure script, it's a hand-crafted one that doesn't support some of the configuration options used by that pipeline: https://github.com/axboe/fio/blob/3ed8eea0ee9b5d7de603e9b128e6c05941b99938/configure#L3-L4

It seems like I could still use `autoconf/make` and `autoconf/make-install` but I'm not sure if that's a good pattern to follow given it's not actually autoconf...

**Review by ajayk — APPROVED** [review]

**Pull request merged**